### PR TITLE
HIVE-24233: Except subquery throws nullpointer

### PR DESCRIPTION
**Pull request by pvargacl**
### What changes were proposed in this pull request?
Throw semanticException for except when cbo is disabled

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
negative q test

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java [hunk 1]
```diff
@@ -11628,6 +11628,11 @@ private void setupStats(TableScanDesc tsDesc, QBParseInfo qbp, Table tab, String
   }
 
   private Operator genPlan(QB parent, QBExpr qbexpr) throws SemanticException {
+    if (qbexpr.getOpcode() == QBExpr.Opcode.EXCEPT || qbexpr.getOpcode() == QBExpr.Opcode.EXCEPTALL
+        || qbexpr.getOpcode() == QBExpr.Opcode.INTERSECT || qbexpr.getOpcode() == QBExpr.Opcode.INTERSECTALL) {
+      throw new SemanticException(
+          "Except and intersect operations are only supported with Cost Based Optimizations enabled. Turn on hive.cbo.enable");
```
Comment (pvary) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java:11634]: Could we say instead or something like it?
```
"EXCEPT and INTERSECT operations are only supported with Cost Based Optimizations enabled. Please set 'hive.cbo.enable' to true"
```

File: ql/src/test/queries/clientnegative/except_cbooff.q [hunk 2]
```diff
@@ -0,0 +1,5 @@
+set hive.cbo.enable=false;
```
Comment (pvary) [inline ql/src/test/queries/clientnegative/except_cbooff.q:1]: Do we have positive tests?

File: ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java (hunk 1 above)
Comment (pvargacl) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java]: Fixed

File: ql/src/test/queries/clientnegative/except_cbooff.q (hunk 2 above)
Comment (pvargacl) [reply ql/src/test/queries/clientnegative/except_cbooff.q:1]: Yes there are, they were added when the original feature was developed.

**Review by pvary — APPROVED** [review]
+1 Pending tests

**Comment by pvargacl** [comment]
@pvary thanks for the review. Could you merge this?

**Pull request merged**